templates/reputation/test/reputation.js

```diff
@@ -39,162 +40,181 @@ contract('Reputation', ([_, owner, holder1, holder2]) => {
+            await assertMissingRole(acl, finance, 'CHANGE_PERIOD_ROLE')
+            await assertMissingRole(acl, finance, 'CHANGE_BUDGETS_ROLE')
+          })
 
-          await assertMissingRole(acl, agent, 'DESIGNATE_SIGNER_ROLE')
-          await assertMissingRole(acl, agent, 'ADD_PRESIGNED_HASH_ROLE')
-        })
+          if(useAgentAsVault) {
+            it('should have agent app correctly setup', async () => {
+              assert.isTrue(await agent.hasInitialized(), 'agent not initialized')
+              assert.equal(await agent.designatedSigner(), ZERO_ADDRESS)
 
-        it('sets up DAO and ACL permissions correctly', async () => {
-          await assertRole(acl, dao, voting, 'APP_MANAGER_ROLE')
-          await assertRole(acl, acl, voting, 'CREATE_PERMISSIONS_ROLE')
-        })
+              assert.equal(await dao.recoveryVaultAppId(), APP_IDS.agent, 'agent app is not being used as the vault app of the DAO')
+              assert.equal(web3.toChecksumAddress(await dao.getRecoveryVault()), agent.address, 'agent app is not being used as the vault app of the DAO')
+
+              await assertRole(acl, agent, voting, 'EXECUTE_ROLE')
+              await assertRole(acl, agent, voting, 'RUN_SCRIPT_ROLE')
+              await assertRole(acl, agent, voting, 'TRANSFER_ROLE', finance)
+
+              await assertMissingRole(acl, agent, 'DESIGNATE_SIGNER_ROLE')
+              await assertMissingRole(acl, agent, 'ADD_PRESIGNED_HASH_ROLE')
+            })
+          }
+          else {
+            it('should have vault app correctly setup', async () => {
+              assert.isTrue(await vault.hasInitialized(), 'vault not initialized')
+
+              assert.equal(await dao.recoveryVaultAppId(), APP_IDS.vault, 'vault app is not being used as the vault app of the DAO')
+              assert.equal(web3.toChecksumAddress(await dao.getRecoveryVault()), vault.address, 'vault app is not being used as the vault app of the DAO')
 
-        it('sets up EVM scripts registry permissions correctly', async () => {
-          const reg = await EVMScriptRegistry.at(await acl.getEVMScriptRegistry())
-          await assertRole(acl, reg, voting, 'REGISTRY_ADD_EXECUTOR_ROLE')
-          await assertRole(acl, reg, voting, 'REGISTRY_MANAGER_ROLE')
+              await assertRole(acl, vault, voting, 'TRANSFER_ROLE', finance)
+            })
+          }
+
+          it('sets up DAO and ACL permissions correctly', async () => {
+            await assertRole(acl, dao, voting, 'APP_MANAGER_ROLE')
+            await assertRole(acl, acl, voting, 'CREATE_PERMISSIONS_ROLE')
+          })
+
+          it('sets up EVM scripts registry permissions correctly', async () => {
+            const reg = await EVMScriptRegistry.at(await acl.getEVMScriptRegistry())
+            await assertRole(acl, reg, voting, 'REGISTRY_ADD_EXECUTOR_ROLE')
+            await assertRole(acl, reg, voting, 'REGISTRY_MANAGER_ROLE')
+          })
```

Comment:
Instead of doing a `for` loop we can move the whole test suite that you wanna ensure for both cases to a function and call it twice, it will look cleaner. I'm thinking sth like
```js
for (const creationStyle of ['single', 'separate']) {
   const itHandlesInstanceCreationsProperly = useAgentAsVault => {
      it('test 1', async () => { ... })
      it('test 2', async () => { ... })
   }
   
   context('when using an agent as vault', () => {
      itHandlesInstanceCreationsProperly(true)
      
      it('should have agent app correctly setup', async () => { ... })
   })

   context('when using a regular vault', () => {
      itHandlesInstanceCreationsProperly(false)

      it('should have vault app correctly setup', async () => {...})
   })
}
```

Reply:
Cool, refactored and implemented this style on all tests.